Source/WebCore/dom/Document.cpp

```diff
@@ -11135,20 +11140,27 @@ void Document::securityOriginDidChange()
 
 #if ENABLE(CONTENT_EXTENSIONS)
 
-ResourceMonitor* Document::resourceMonitor()
+ResourceMonitor* Document::resourceMonitorIfExists()
 {
     return m_resourceMonitor.get();
 }
 
-void Document::setResourceMonitor(RefPtr<ResourceMonitor>&& resourceMonitor)
+ResourceMonitor& Document::resourceMonitor()
+{
+    if (!m_resourceMonitor)
+        m_resourceMonitor = ResourceMonitor::create(*frame());
+    return *m_resourceMonitor.get();
+}
+
+Ref<ResourceMonitor> Document::protectedResourceMonitor()
 {
-    m_resourceMonitor = WTFMove(resourceMonitor);
+    return resourceMonitor();
 }
 
 ResourceMonitor* Document::parentResourceMonitor()
```

Comment:
Should use `IfExists` since that's the pattern you're adopting elsewhere in your PR (though I personally prefer the `T* foo()` / `T& ensureFoo()` style personally). At the very least you should be consistent with yourself.

Reply:
But it is documented so I should follow the guideline. :)